test/apps/widgets/layer-list-widget.ts

```diff
@@ -0,0 +1,108 @@
+import type {Viewport, WidgetPlacement, Layer} from '@deck.gl/core';
+import {_WidgetImpl, WidgetImplProps} from '@deck.gl/widgets';
+import {h, Fragment, render} from 'preact';
+
+type LayerListWidgetProps = {
+  id?: string;
+  /**
+   * Widget positioning within the view. Default: 'top-left'.
+   */
+  placement?: WidgetPlacement;
+  /**
+   * View to attach to and interact with. Required when using multiple views. Default: null
+   */
+  viewId?: string | null;
+  /**
+   * CSS inline style overrides.
+   */
+  style?: Partial<CSSStyleDeclaration>;
+  /**
+   * Additional CSS class.
+   */
+  className?: string;
+};
+
+export class LayerListWidget extends _WidgetImpl<LayerListWidgetProps> {
+  static defaultProps: Required<LayerListWidgetProps> = {
+    ..._WidgetImpl.defaultProps,
+    id: 'layer-list',
+    placement: 'top-left'
+  };
+
+  // WidgetImpl
+  className = 'deck-widget-layer-list';
+  // IWidget
+  placement: WidgetPlacement = 'top-left';
+  viewId?: string | null = null;
+  // Custom
+  viewports: {[id: string]: Viewport} = {};
+  layers: Layer[] = [];
+
+  constructor(props: LayerListWidgetProps = {}) {
+    super({...LayerListWidget.defaultProps, ...props});
+    this.placement = props.placement ?? this.placement;
+    this.viewId = props.viewId ?? this.viewId;
+  }
+
+  setProps(props: Partial<LayerListWidgetProps>) {
+    this.placement = props.placement ?? this.placement;
+    this.viewId = props.viewId ?? this.viewId;
```

Comment:
Call out: The widget system expects properties like `placement` and `viewId` to be assigned at the top-level. Not all widgets would expose these as props.